pages/_app.js

```diff
@@ -51,6 +55,27 @@ function App({ Component, pageProps }) {
     };
   }, [router.events]);
 
+  useEffect(() => {
+    // Do not run Amplitude Analytics unless traffic is coming from a browser
+    const _isNotBrowser =
+      typeof window === 'undefined' || typeof document === 'undefined';
+
+    if (_isNotBrowser) return null;
+
+    // Only run Amplitude Analytics in production
+    if (!process.env.NODE_ENV === 'production') return null;
+
+    // NEXT_PUBLIC_AMPLITUDE_KEY  needs to be set in the .env
+    if (!process.env.NEXT_PUBLIC_AMPLITUDE_KEY) {
+      console.warn(
+        'Amplitude Analytics tracking code is required to initialize Amplitude Analytics'
+      );
+      return null;
+    }
+
+    amplitude.init(currentUser);
```

Comment:
What about unauthenticated users?
Do you think we need to do any special handling, or pass `null` or something to allow a unique one to be generated for this session? 🤔

Reply:
Yeah, definitely need to take that in to consideration 👍